Assets/Scripts/BlockScripts/nextLetter.cs

```diff
@@ -4,26 +4,32 @@
 public class NextLetter : MonoBehaviour
 {
 
-    public static char GetNextLetter(List<Letter> alphabet){
-        int totalWeight = 0; 
-        foreach(var letter in alphabet)
+    public static char GetNextLetter(List<Letter> alphabet)
+    {
+        int totalWeight = 0;
+        char val = new char();
```

Comment:
You can set val to be '\0' from the get go, so if randomNumber is never less then letter.weight then it will return '\0' by default

Reply:
good shout